.github/workflows/integration_test_charm_self_hosted.yaml

```diff
@@ -0,0 +1,236 @@
+              print("Skipping unstable tests")
+              output = "mark_expression=not unstable"
+
+          with open(os.environ["GITHUB_OUTPUT"], "a") as file:
+              file.write(output)
+      - name: Collect test groups
+        id: collect-groups
+        run: tox run -e integration -- tests/integration -m '${{ steps.select-test-stability.outputs.mark_expression }}' --collect-groups
+    outputs:
+      groups: ${{ steps.collect-groups.outputs.groups }}
+
+  integration-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        groups: ${{ fromJSON(needs.collect-integration-tests.outputs.groups) }}
+    name: ${{ matrix.groups.job_name }}
+    needs:
+      - get-workflow-version
+      - collect-integration-tests
+    runs-on: [self-hosted, linux, X64, jammy, large]
+    timeout-minutes: 120
+    steps:
+      - name: Install CLI
+        run: |
+          sudo apt-get install python3-pip python3-venv -y
+          python3 -m pip install --user pipx
+          python3 -m pipx ensurepath
+          pipx install git+https://github.com/canonical/data-platform-workflows@'${{ needs.get-workflow-version.outputs.version }}'#subdirectory=python/cli
+      - name: Redact secrets from log
+        run: redact-secrets
+        env:
+          SECRETS: ${{ secrets.integration-test }}
+      - name: Validate cloud input
+        shell: python
+        # Keep synchronized with inputs.cloud description
+        run: assert "${{ inputs.cloud }}" in ("lxd")
+      - name: Parse Juju agent version & snap channel
+        id: parse-versions
+        shell: python
+        run: |
+          import os
+
+          AGENT_VERSION = "${{ inputs.juju-agent-version }}"
+          snap_channel = "${{ inputs.juju-snap-channel }}"
+          if not snap_channel:
+              if AGENT_VERSION:
+                  major, minor, patch = AGENT_VERSION.split(".")
+                  snap_channel = f"{major}.{minor}/stable"
+              else:
+                  raise Exception(
+                      "`juju-snap-channel` required if `juju-agent-version` is not passed"
+                  )
+
+          output = "agent_bootstrap_option="
+          if AGENT_VERSION:
+              output += f"--agent-version={AGENT_VERSION}"
+          output += f"\nsnap_channel={snap_channel}"
+
+          print(output)
+          with open(os.environ["GITHUB_OUTPUT"], "a") as file:
+              file.write(output)
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up environment
+        run: |
+          # `--classic` applies to juju 2 snap; ignored for juju 3 snap
+          sudo snap install juju --classic --channel='${{ steps.parse-versions.outputs.snap_channel }}'
+      - name: Set up lxd
+        if: ${{ inputs.cloud == 'lxd' }}
+        run: |
+          lxd waitready
+          lxd init --auto
+          lxc network set lxdbr0 ipv6.address none
+          sudo iptables -F FORWARD
+          sudo iptables -P FORWARD ACCEPT
```

Comment:
JFYI only, I am not sure it is still necessary... I do not see it in [Multipass blueprint](https://github.com/sed-i/multipass-workflows/blob/3746f53be4e9b7b53692fe58dde8ae72af5a8877/v1/charm-dev.yaml), but let's keep it here.